### PR TITLE
[ pattgen ] Transition to D2

### PR DESCRIPTION
**Pull request by martin-lueker**
Modulo a few questions about CDC signoff and FPGA Area, this block is ready to advance to D2

Signed-off-by: Martin Lueker-Boden <martin.lueker-boden@wdc.com>

### CODE REVIEW

**Comment by martin-lueker** [comment]
FYI @mateenfaisal

File: hw/ip/pattgen/doc/checklist.md
Comment (msfschaffner) [inline hw/ip/pattgen/doc/checklist.md:45]: You can delete this line. It has been superseded by `AREA_CHECK` above.
Please also delete the line `[AREA_CROSS_CHECK]:    {{<relref "/doc/project/checklist.md#area_cross_check" >}}` below.

@imphil OTBN seems to have the same duplicate in the checklist.

File: hw/ip/pattgen/doc/checklist.md
Comment (msfschaffner) [inline hw/ip/pattgen/doc/checklist.md:51]: you can write `| waived | CDC flow is not available yet.` into the last two columns.
see here for an example:
https://docs.opentitan.org/hw/ip/sram_ctrl/doc/checklist/index.html#d2

File: hw/ip/pattgen/data/pattgen.prj.hjson
Comment (msfschaffner) [inline hw/ip/pattgen/data/pattgen.prj.hjson:16]: @cfrantz @mateenfaisal 
Who is the SW owner of this DIF?
It would be good if someone from the SW side could have a look at the CSRs before we move this into D2...

File: hw/ip/pattgen/doc/checklist.md
Comment (msfschaffner) [inline hw/ip/pattgen/doc/checklist.md:52]: This currently meets timing on FPGA, so you can mark this as `Done`.

File: hw/ip/pattgen/doc/checklist.md
Comment (msfschaffner) [inline hw/ip/pattgen/doc/checklist.md:53]: Does the code use/need sync macros at all? If not, this should be `N/A`.

**Review by msfschaffner — COMMENTED** [review]
Thanks @martin-lueker - see comments below. You may also have to rebase this onto master to include some of the latest CI fixes.

File: hw/ip/pattgen/data/pattgen.prj.hjson
Comment (mateenfaisal) [reply hw/ip/pattgen/data/pattgen.prj.hjson:16]: afaik, we don't have a SW owner for PATTGEN DIF but someone from SW side must look into it. @asb/@cfrantz   - can you please allocate a person from SW team to review PATTGEN CSR?

File: hw/ip/pattgen/doc/checklist.md
Comment (imphil) [reply hw/ip/pattgen/doc/checklist.md:45]: Thanks @msfschaffner, I updated OTBN in https://github.com/lowRISC/opentitan/pull/6809.

File: hw/ip/pattgen/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/pattgen/doc/checklist.md:53]: Hi @msfschaffner, Right, I did this and PWM at the same time.   And there are no CDC's here.  Will change to N/A.

File: hw/ip/pattgen/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/pattgen/doc/checklist.md:52]: - Thanks.

File: hw/ip/pattgen/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/pattgen/doc/checklist.md:45]: Will do.

**Comment by martin-lueker** [comment]
So is the consensus that we are going to wait on a SW review on this?

**Comment by msfschaffner** [comment]
> So is the consensus that we are going to wait on a SW review on this?

Let's wait for SW. 
Both this one and PWM are relatively simple IPs, so SW review should not take too long hopefully...

File: hw/ip/pattgen/data/pattgen.prj.hjson
Comment (cfrantz) [reply hw/ip/pattgen/data/pattgen.prj.hjson:16]: @mcy agreed to review the PATTGEN doc and register spec.

Comment (msfschaffner) [reply hw/ip/pattgen/data/pattgen.prj.hjson:16]: Great, thanks!

**Review by mcy — COMMENTED** [review]
So just to ensure that I understand what this device does:

> For each channel, I can send N copies of a string of M bits, where each bit is sent at a specified rate (relative to the IO clock).

If so, I think the registers are pretty reasonable. One nit that I would make is that it may be worth specifying the behavior of `ENABLE_CH#` after the pattern is complete. From reading it, it sounds like it will remain high, which admittedly feels a little odd, given that presumably the channel's clock will have stopped.

**Review by imphil — APPROVED** [review]
LGTM. 

@martin-lueker can you do a follow-up PR to address the documentation question @mcy had? (Can go in after the D2 merge.)

**Review by msfschaffner — APPROVED** [review]

**Comment by martin-lueker** [comment]
Great. Thanks everyone.  @mcy's regarding your request: It shouldn't be too hard. 

@imphil: If you are willing to merge this one let me at least get an issue open.  I don't want this to go in as D2, and then have that change fall behind the cracks.

**Review by tunghoang290780 — APPROVED** [review]

**Comment by mateenfaisal** [comment]
Hey @martin-lueker - I was going to open a GH issue (that mcy raised) to remove the hurdle in the merging of pattgen IP D2 PR  :). 

But then I stopped because you have already opened an issue (that mcy raised). :)

**Comment by imphil** [comment]
@martin-lueker Oh I don't mind (and actually prefer) merging this PR only when all open issues are resolved. Really up to you.

**Comment by martin-lueker** [comment]
Created new issue #7214 to handle @mcy's request.

**Pull request merged**
